### PR TITLE
project structure suggestion for multiple optimizers

### PR DESCRIPTION
**Pull request by sebasv**
@to266 I'm open to suggestions, let me know what you think of the separate scalar and vector modules.

### CODE REVIEW

File: src/scalar/mod.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+//! This module contains all minimizers for functions with a single argument.
```
Comment (to266) [inline src/scalar/mod.rs]: "single argument" may be a bit misleading, don't you think? How about something along the lines of "Minimizers that minimize functions along one dimension"? This particular sentence also does not "flow", but the main idea is that the number of arguments is not really constrained, it's just that the minimizer will only tweak one "knob" to find the minimum.

File: src/vector/mod.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+//! This module contains all minimizers for functions with multiple arguments.
```
Comment (to266) [inline src/vector/mod.rs]: Similarly, I would suggest something along the lines of "will minimize the function value across multiple dimensions"

File: src/scalar/mod.rs (hunk 1 above)
Comment (sebasv) [reply src/scalar/mod.rs]: Agreed, the current formulation is a bit vague. I'll push some changes!

**Comment by sebasv** [comment]
I've given more accurate descriptions in the module docstrings. As more methods are added, we might put some guidance in there as well to help unfamiliar users choose minimizers that fit their problem.

**Comment by to266** [comment]
Excellent!

**Pull request merged**